components/governance/org.wso2.carbon.governance.taxonomy/src/main/java/org/wso2/carbon/governance/taxonomy/internal/TaxonomyServiceComponent.java

```diff
@@ -43,15 +63,63 @@
      *
      * @param context OSGi component context.
      */
-    protected void activate(ComponentContext context) {
+    protected void activate(ComponentContext context)
+            throws UserStoreException, RegistryException, ParserConfigurationException, SAXException,
+            XPathExpressionException, IOException {
+
         BundleContext bundleContext = context.getBundleContext();
-        bundleContext.registerService(
-                TaxonomyService.class.getName(), new TaxonomyTreeAPI(), null);
+        TenantLoginStorageService tenantLoginStorageService = new TenantLoginStorageService();
+        ServiceRegistration tenantMgtListenerSR = bundleContext
+                .registerService(Axis2ConfigurationContextObserver.class.getName(), tenantLoginStorageService, null);
+
+        if (tenantMgtListenerSR != null) {
+            if (log.isDebugEnabled()) {
+                log.debug("Governance Taxonomy Management - TenantLoginStorageService registered");
+            }
+        } else {
+            log.error("Governance Taxonomy Management - TenantLoginStorageService could not be registered");
+        }
+
+        BundleContext taxonomyBundleContext = context.getBundleContext();
+        ServiceRegistration taxonomyManagementService = taxonomyBundleContext
+                .registerService(ITaxonomyServices.class.getName(), new TaxonomyServicesImpl(), null);
+
+        if (taxonomyManagementService != null) {
+            if (log.isDebugEnabled()) {
+                log.debug("Governance Taxonomy Management - Manager services registered");
+            }
+        } else {
+            log.error("Governance Taxonomy Management - Manager services could not be registered");
+        }
 
         if (log.isDebugEnabled()) {
-            log.debug("Taxonomy bundle is activated");
+            log.debug("Governance Taxonomy Management Service bundle is activated");
+        }
+
+        try {
+            TaxonomyManager taxonomyManager = new TaxonomyManager();
+            taxonomyManager.initTaxonomyStorage();
+            if (log.isDebugEnabled()) {
+                log.debug("Governance Taxonomy Management map initialization is success");
+            }
+        } catch (UserStoreException e) {
+            log.error("Error occurred while getting RealmConfigurations when activating osgi service", e);
+        } catch (RegistryException e) {
+
+            log.error("Error occurred while getting taxonomy files from registry when activating osgi service", e);
+        } catch (IOException e) {
+
+            log.error("Error occurred while parsing taxonomy xml when activating osgi service", e);
+        } catch (ParserConfigurationException e) {
+
+            log.error("Error occurred while building new document for taxonomy when activating osgi service", e);
+        } catch (SAXException e) {
+
```

Comment:
Thank you Hasitha for mentioning it,
this fixed with https://github.com/wso2/carbon-governance/pull/256/commits/677bdff1879f1d46104c09434267cecf2a92d086